### PR TITLE
Updates to AlertDialog

### PR DESCRIPTION
**Pull request by BowerSteve**
#### WHAT
Add AlertDialog overload for variation with stack of Chips and no Ok/Cancel buttons.

* Standardised onOk, onCancel parameter names
* Added content parameter to AlertDialog for additional scrollable content
* message parameter no longer restricted to 3 lines

#### WHY
Developers should now be able to use AlertDialog and Confirmation for all responsive dialogs, without calling the low-level ResponsiveDialogContent 

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt [hunk 1]
```diff
@@ -32,59 +33,100 @@ import com.google.android.horologist.compose.layout.rememberColumnState
...
 
+/**
+ * This component is an alternative to [AlertContent], providing the following:
+ * - a convenient way of passing a title and a message;
+ * - slot for scrollable content (including stack of Chips for options);
+ * - wrapped in a [Dialog];
+ */
+@ExperimentalHorologistApi
+@Composable
+public fun AlertDialog(
```
Comment (yschimke) [inline compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt:86]: Do you need this one? Or are you worried about backwards compatibility?

If so, I suggest just changing the API and dropping this one.

File: sample/src/test/snapshots/images/com.google.android.horologist.screensizes_DialogTest_batterySaverScreen[0]_mobvoiticwatchpro5.png
Comment (yschimke) [inline sample/src/test/snapshots/images/com.google.android.horologist.screensizes_DialogTest_batterySaverScreen[0]_mobvoiticwatchpro5.png:1]: Are you happy with left aligned on these?

File: compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt (hunk 1 above)
Comment (BowerSteve) [reply compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt:86]: The overload requires (defaulted) content descriptions for ok/cancel buttons, even if those buttons aren't part of the dialog. So I think it's clearer to have a separate overload without those parameters at all.

Comment (yschimke) [reply compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt:86]: OK.

File: sample/src/test/snapshots/images/com.google.android.horologist.screensizes_DialogTest_batterySaverScreen[0]_mobvoiticwatchpro5.png
Comment (BowerSteve) [reply sample/src/test/snapshots/images/com.google.android.horologist.screensizes_DialogTest_batterySaverScreen[0]_mobvoiticwatchpro5.png:1]: Good spot. I think message should be centered - and if developers want left-aligned text, they can do it using the content parameter.

File: sample/src/test/snapshots/images/com.google.android.horologist.screensizes_DialogTest_longDialogScreen1[1]_samsunggalaxywatch5.png
[the remark this answers is not in the document]
Comment (BowerSteve) [reply sample/src/test/snapshots/images/com.google.android.horologist.screensizes_DialogTest_longDialogScreen1[1]_samsunggalaxywatch5.png:1]: Have specified onBackground as the color, which looks right even specs. We could add color parameters for title and message if needed.

File: compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt [hunk 2]
```diff
@@ -101,19 +157,34 @@ public fun AlertContent(
...
+                // Should message be start or center aligned?
+                val textMeasurer = rememberTextMeasurer()
+                val textStyle = LocalTextStyle.current
+                val lineCount = remember(it, textStyle, textMeasurer) {
+                    textMeasurer.measure(
+                        text = it,
+                        style = textStyle,
+                        constraints = Constraints(
+                            // Available width is reduced by responsive dialog horizontal padding.
+                            maxWidth = (maxScreenWidthPx * messageMaxWidthFraction).toInt(),
```
Comment (ssancho14) [inline compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt:169]: This is counting the  4.16% extra adding for message but not the global 5.2% on the whole Dialog content

Comment (BowerSteve) [reply compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt:169]: The global 5.2% is now in the 'calculatePaddingFraction' function - please check.

File: compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt [hunk 3]
```diff
@@ -101,19 +157,34 @@ public fun AlertContent(
         },
         message = message?.let {
             {
+                // Should message be start or center aligned?
+                val textMeasurer = rememberTextMeasurer()
+                val textStyle = LocalTextStyle.current
+                val lineCount = remember(it, textStyle, textMeasurer) {
+                    textMeasurer.measure(
```
[the remark this answers is not in the document]
Comment (BowerSteve) [reply compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt:164]: Good idea, will add it.

File: compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt (hunk 2 above)
Comment (BowerSteve) [reply compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt]: Updated the calculation in AlertDialog.

File: compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (ssancho14) [reply compose-material/src/main/java/com/google/android/horologist/compose/material/AlertDialog.kt:165]: ?

**Review by ssancho14 — APPROVED** [review]

**Pull request merged**